surface/include/pcl/surface/impl/gp3.hpp

```diff
@@ -878,10 +878,10 @@ pcl::GreedyProjectionTriangulation<PointInT>::reconstructPolygons (std::vector<p
           else
             angle_so_far = 0;
         }
-        for (const int &it : to_erase)
+        for (const auto &idx : to_erase)
         {
           for (std::vector<int>::iterator iter = angleIdx.begin(); iter != angleIdx.end(); ++iter)
```

Comment:
Range loop?

Reply:
Not possible, because `iter` is needed for `erase` in line 884.